### PR TITLE
Enfore optional braces

### PR DESCRIPTION
**Pull request by gunchleoc**
- Add remaining optional braces
- Enforce optional braced by running clang-tidy on Travis

Sample error output:

```
#######################################
#######################################
###                                 ###
###   Checking clang-tidy results   ###
###                                 ###
#######################################
#######################################
/home/travis/build/widelands/widelands/src/economy/ship_fleet.cc:83:7: warning: statement should be inside braces [google-readability-braces-around-statements]
/home/travis/build/widelands/widelands/src/economy/test/test_routing.cc:51:33: warning: statement should be inside braces [google-readability-braces-around-statements]
/home/travis/build/widelands/widelands/src/economy/test/test_routing.cc:140:22: warning: statement should be inside braces [google-readability-braces-around-statements]
/home/travis/build/widelands/widelands/src/logic/map_objects/tribes/warehouse.cc:392:6: warning: statement should be inside braces [google-readability-braces-around-statements]
#######################################
#######################################
###                                 ###
###   Found 4 error(s)             ###
###                                 ###
#######################################
#######################################
```

### CODE REVIEW

**Comment by gunchleoc** [comment]
This branch is now ready for review

File: .gitignore [hunk 1]
```diff
@@ -24,5 +24,6 @@ VERSION
 # QT Creator settings
 CMakeLists.txt.user*
 
-# Translation validation files
+# Validation checks
 po_validation
+/clang-tidy.log
```
Comment (Noordfrees) [inline .gitignore]: delete leading /

File: utils/check_clang_tidy_results.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+    errors = 0
+
+    with open(log_file) as checkme:
+        contents = checkme.readlines()
+        for line in contents:
+            if 'third_party' in line:
+                continue
+            check_suppressed = False
+            for check in SUPPRESSED_CHECKS:
```
Comment (Noordfrees) [inline utils/check_clang_tidy_results.py:132]: This SUPPRESSED_CHECKS design looks rather fragile to me. Wouldn't it be better to instead filter for a brief list of checks which we want?

File: .travis.yml [hunk 3]
```diff
@@ -57,6 +57,28 @@ jobs:
       language: python
       script: ./.travis.sh includes
       cache: false
+    - name: "Clang-tidy"
```
Comment (Noordfrees) [inline .travis.yml]: This check seems to take very long on travis. Please move it to the second build stage for efficiency.

**Review by Noordfrees — CHANGES_REQUESTED** [review]

File: utils/check_clang_tidy_results.py (hunk 2 above)
Comment (gunchleoc) [reply utils/check_clang_tidy_results.py:132]: That's is the design I had at first, but this way around it is a lot more maintainable,e.g. when we want to clear more checks, we already have a ready-made list of checks that are available.

**Comment by Noordfrees** [comment]
The check name on travis now needs fixing. Otherwise LGTM.

**Review by Noordfrees — APPROVED** [review]

**Comment by gunchleoc** [comment]
AppVeyor failure is a failed upload. Merging.

**Pull request merged**